### ISSUE TITLE
[V2] Cannot modify an existing edge deployment's modules and routes

### ISSUE THREAD

**Issue by leeprtc** [Azure/iot-edge#509]
Currently, you can only modify an existing edge deployment's target condition, labels, and priority. You cannot modify the modules and routes for an edge deployment.

Since the definition for modules are before the definition for routes, it is easy to have typos when defining the routes. The user experience will be better if you can also change the modules and routes of an edge deployment. I'm ok if you restrict the changes to just the routes. Without this ability to modify, one has to delete and then re-create the edge deployment.

**Comment by mrohera** [comment]
Conceptually, a deployment should be thought of as a unit (or package) comprising the set of modules, their respective configurations and routes between these. Updates to any of these items implies a new deployment push requested by the user. I can see your point about being a bit challenging to use during development but on the flip side, this is precisely what is useful to customers who would like to control exactly what is deployed on their edge device at any point in time.

I have assigned your query to @chrissie926 from our UX team who can possibly comment further on your UX feedback.

Please let us know if you have any other questions and thank you for your feedback.

**Comment by leeprtc** [comment]
Can you create a new deployment based on an existing deployment?

The thinking here is if you can use the existing deployment to fill in the deployment info and make everything editable, then the same goal is achieved while still preserving the semantics of new deployment if anything changes.

**Comment by mrohera** [comment]
Yup, 'cloning' a deployment and editing is a very logical option but I will let @chrissie926 chime in.
